pkg/validation/spec/contact_info.go

```diff
@@ -18,7 +18,7 @@ package spec
 //
 // For more information: http://goo.gl/8us55a#contactObject
 type ContactInfo struct {
-	Name  string `json:"name,omitempty"`
-	URL   string `json:"url,omitempty"`
-	Email string `json:"email,omitempty"`
+	Name  string `json:"name,omitempty" yaml:"name,omitempty"`
+	URL   string `json:"url,omitempty" yaml:"url,omitempty"`
+	Email string `json:"email,omitempty" yaml:"email,omitempty"`
```

Comment:
I _really_ don't think we should start adding yaml tags to our API types. Having benchmarked the yaml decoding in the past, I'm also really surprised it is faster than the json decoding (it was generally significantly slower in the past). Do we know which bits the json decoding is super slow on?

Reply:
Yaml v3 package since 2019 operates on an AST; did you look before then? 

My benchmarks measured unmarshalling from the AST type instead of text. This is sufficient for my use case of providing a path to convert from protobuf: `protoubf -> google/gnostic -> kube-openapi`

I haven't benchmarked it, but I expect going from YAML test -> AST -> Kube-Openapi would be comparable/slower than JSON.

Reply:
If we want a google/gnostic → kube-openapi path, it would make more sense to build that transformation and round-trip test it to me. Using yaml AST bits is clever, but I don't think we should decorate API types with yaml tags and push consumers in that direction

Reply:
> I don't think we should decorate API types with yaml tags and push consumers in that direction

I'm curious why you're against having yaml tags here? The proto->gnostic->kube-openapi conversion that this facilitates is a critical performance improvement that kpt and kustomize would like to have as soon as possible.

Reply:
We have an alternative solution that is probably going to be even faster, but it will take a few weeks to implement. This is also a blocker for next code-freeze so we're trying to move fast on that!

Reply:
Thank you so much for the update!